Lib/test/test_os.py

```diff
@@ -4503,6 +4503,23 @@ def test_times(self):
             self.assertEqual(times.elapsed, 0)
 
 
+@requires_os_func('fork')
+@requires_os_func('waitpid')
+class ForkTests(unittest.TestCase):
+    def test_fork(self):
+        # bpo-42540: ensure os.fork() with non-default allocator
+        code = """if 1:
+            import os
+            pid = os.fork()
+            if pid != 0:
+                _, exitcode = os.waitpid(pid, 0)
+                assert exitcode == 0
+        """
+        assert_python_ok("-c", code)
+        if support.with_pymalloc():
+            assert_python_ok("-c", code, PYTHONMALLOC="pymalloc_debug")
```

Comment:
If you want to test a non-default allocator, you can use PYTHONMALLOC="malloc_debug" (which doesn't need with_pymalloc() context manager). pymalloc_debug is the default for a debug build.